Automattic/jetpack
# Show product name on activation notice

Fixes 1164141197617539-as-1201858908716168

#### Changes proposed in this Pull Request:

This PR includes the following changes

* Add an API to get all the licenses 
* Show product name in the activation notice if only one product is available to be activated
* Fixed one issue where it was not showing the activation notice when 
  - You have 2 licenses to activate
  - Activate 1 license
  - Now you have 1 license to activate, but it was not showing the activation notice. 
  
  Code change to fix that issue
  
  ```const userHasNewDetachedLicenses = detachedLicensesCount > ( lastDetachedCount || 0 );``` 
  
  to 
  
  ```const userHasNewDetachedLicenses = detachedLicensesCount >= ( lastDetachedCount || 0 );```


#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions

1. Setup a new test site using https://jurassic.ninja/specialops/ with this branch.
2. Visit the newly create site -> Jetpack -> Activate Jetpack.
3. Go to https://jetpack.com/ and buy any product and skip the site selection.
4.  Now, go to your site -> Jetpack(refresh if you are on the same page) and you should see the activation notice with the message as `Activate <PRODUCT_NAME> ...`
5. Try buying another plan by repeating Step 3 and Step 4, you should show see the activation notice with the message as `You have an available product license key. ...`  
6. Now try activating any product available by clicking on Activate Now on the activation notice. You can get the activation key from here - https://wordpress.com/me/purchases where you can also see the products which are not activated. You should now see the message as `Activate <PRODUCT_NAME> ...` [Repeat Step 4]
7. Now, activate the other product and repeat step 4, you should not see any activation notice now.

Jetpack dashboard before any available product for activation

<img width="1037" alt="Screenshot 2022-03-15 at 12 02 39 PM" src="https://user-images.githubusercontent.com/10586875/158322145-901ed76f-ac21-43d3-a02c-c7a72c093b36.png">

Jetpack dashboard when there is only one available product for activation

<img width="1039" alt="Screenshot 2022-03-15 at 11 59 41 AM" src="https://user-images.githubusercontent.com/10586875/158322230-e0d120ed-2ba9-493f-9246-7367dc35738f.png">

Jetpack dashboard when there is more than one available product for activation

<img width="1033" alt="Screenshot 2022-03-15 at 12 00 48 PM" src="https://user-images.githubusercontent.com/10586875/158322292-f6b2a673-48ba-4ca3-977b-38e5a9a6092c.png">
